akka-actor/src/main/scala/akka/io/dns/DnsSettings.scala

```diff
@@ -25,13 +25,20 @@ private[dns] final class DnsSettings(system: ExtendedActorSystem, c: Config) {
   import DnsSettings._
 
   val NameServers: List[InetSocketAddress] = {
-    val addrs = Try(c.getString("nameservers")).toOption.toList
-      .flatMap {
-        case "default" ⇒ getDefaultNameServers(system).getOrElse(failUnableToDetermineDefaultNameservers)
-        case address   ⇒ parseNameserverAddress(address) :: Nil
-      }
-    if (addrs.nonEmpty) addrs
-    else c.getStringList("nameservers").asScala.map(parseNameserverAddress)(breakOut)
+    val addresses = Try(c.getString("nameservers")).toOption match {
```

Comment:
I think using `c.getValue("nameservers")` and matching on the `ConfigValue` sub type is nicer than doing it like this.

Reply:
i thought i tried this and the types are package private

Reply:
Nope, `com.typesafe.config.{ConfigString,ConfigList}` etc are public types.

Reply:
Oh, you are right, hmm, could I use them elsewhere then?

Reply:
Ah, there is an enum, `ConfigValueType`, you can get that from `ConfigValue.valueType`

Reply:
ah that works, thanks for the tip!

Reply:
Updated to use the enum